docs/source/daml2ts/index.rst

```diff
@@ -0,0 +1,291 @@
+
+The DAML type ``Pair a b`` has been modeled in Typescript as a discriminated union with a single case. We'll see more discriminated unions in the next section.
+
+Variants
+~~~~~~~~
+
+Basic
+*****
+
+This DAML example defines a type for the language of additive expressions.
+
+.. code-block:: daml
+   :linenos:
+
+   data Expr =
+       Lit Int
+     | Var Text
+     | Add (Expr, Expr)
+
+In Typescript, the type is represented as a discriminated union. The ``Add`` branch of the type utilizes a 2-element tuple.
+
+.. code-block:: typescript
+   :linenos:
+
+   import * as pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types from './../40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7/DA/Types';
+
+   export type Expr =
+     |  { tag: 'Lit'; value: daml.Int }
+     |  { tag: 'Var'; value: string }
+     |  { tag: 'Add'; value: pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2<Expr, Expr> }
+
+   export const Expr:
+     daml.Serializable<Expr> & {
+     } = ({
+     decoder: () => jtv.oneOf<Expr>(
+       jtv.object({tag: jtv.constant('L'), value: jtv.lazy(() => daml.Int.decoder())}),
+       jtv.object({tag: jtv.constant('V'), value: jtv.lazy(() => daml.Text.decoder())}),
+       jtv.object({tag: jtv.constant('A'), value: jtv.lazy(() => pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2(Expr, Expr).decoder())}),
+     ),
+   });
+
+Generic
+*******
+
+Suppose we tweak the above expression type so that it is polymorphic in the type of literals.
+
+.. code-block:: daml
+   :linenos:
+
+   data Expr a =
+       Lit a
+     | Var Text
+     | Add (Expr a, Expr a)
+
+The resulting Typescript will then generalize to this.
+
+.. code-block:: typescript
+   :linenos:
+
+   import * as pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types from './../40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7/DA/Types';
+
+   export type Expr<a> =
+     |  { tag: 'Lit'; value: a }
+     |  { tag: 'Var'; value: string }
+     |  { tag: 'Add'; value: pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2<Expr<a>, Expr<a>> }
+
+   export const Expr = <a>(a: daml.Serializable<a>): daml.Serializable<Expr<a>> => ({
+     decoder: () => jtv.oneOf<Expr<a>>(
+       jtv.object({tag: jtv.constant('Lit'), value: jtv.lazy(() => a.decoder())}),
+       jtv.object({tag: jtv.constant('Var'), value: jtv.lazy(() => daml.Text.decoder())}),
+       jtv.object({tag: jtv.constant('Add'), value: jtv.lazy(() => pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2(Expr(a), Expr(a)).decoder())}),
+     ),
+   });
+
+Sum-of-products
+~~~~~~~~~~~~~~~~
+Foo.
+
+Basic
+*****
+Bar.
+
+Generic
+*******
+Baz.
+
+Enums
+~~~~~
+Foo, bar, baz.
+
+Templates and choices
+~~~~~~~~~~~~~~~~~~~~~
+Foo, bar, baz.
```

Comment:
This is missing the most important part for me as a user. How do I invoke it and include the generated code in my project?

Reply:
It's coming. Documenting how to invoke `daml2ts` is a pending task - first we have to get it behind the assistant. For now, the focus here is on just explaining what it generates.